scripts/generate_errors.pl

```diff
@@ -193,6 +245,110 @@
 $error_format =~ s/LOW_LEVEL_CODE_CHECKS\n/$ll_code_check/g;
 $error_format =~ s/HIGH_LEVEL_CODE_CHECKS\n/$hl_code_check/g;
 
-open(ERROR_FILE, ">$error_file") or die "Opening destination file '$error_file': $!";
-print ERROR_FILE $error_format;
-close(ERROR_FILE);
+open(ERROR_FILE, ">$error_c") or die "Opening destination file '$error_c': $!";
+print ERROR_FILE $error_format or die "Writing '$error_c': $!";
+close(ERROR_FILE) or die "Closing '$error_c': $!";
+
+sub check_range
+{
+    return if @_ <= 3;
+    my $name = shift @_;
+    my $min = shift @_;
+    my $max = pop @_;
+    foreach my $x (@_)
+    {
+        if ($x < $min || $x > $max)
+        {
+            printf STDERR ("%s (0x%04x) out of range 0x%04x-0x%04x\n",
+                           $name, $x, $min, $max);
+            exit 1;
+        }
+    }
+}
+
+my %h_low_entries;
+my %h_high_entries;
+foreach my $value (keys %error_codes_seen)
+{
+    my $name = $error_codes_seen{$value};
+    if ($value <= 0x7f)
+    {
+        $h_low_entries{$name} = {nr=>0, even=>[], odd=>[]}
+            unless exists $h_low_entries{$name};
+        ++$h_low_entries{$name}{nr};
+        push @{$h_low_entries{$name}{($value & 1 ? 'odd' : 'even')}}, $value;
+    }
+    else
+    {
+        if ($name eq 'SSL' && $value >= 0x7000)
+        {
+            # Hack because SSL occupies two high-level module IDs
+            $name = 'SSL ';
+        }
+        $h_high_entries{$name} = {nr=>0, codes=>[]}
+            unless exists $h_high_entries{$name};
+        ++$h_high_entries{$name}{nr};
+        push @{$h_high_entries{$name}{codes}}, $value;
+    }
+}
+foreach my $name (keys %h_low_entries)
+{
+    my $entry = $h_low_entries{$name};
+    my @even = sort {$a <=> $b} @{$entry->{even}};
+    my @odd = sort {$a <=> $b} @{$entry->{odd}};
+    check_range($name, @even);
```

Comment:
I might be missing something obvious, but I don't get it: you just sorted that array, so how could `check_range()` fail?